lib/avalon/batch/entry.rb

```diff
@@ -160,8 +160,10 @@ def process!
           master_file.set_workflow(file_spec[:skip_transcoding] ? 'skip_transcoding' : nil)
           save_tries = 0
           begin
-            saved = media_object.save
+            media_object.save
+            saved = true
```

Comment:
Actually, now that I look deeper, I think that the original code was better and this patch should be reverted -- it would allow invalidated media objects to get processed (bad).

The problem here (surprise) is that the language is not getting set properly for the media object used in the test, so a validation is making the test fail, not the retry code. It's likely that some mock data for the test needs to be modified to inject the language in.